### PR TITLE
add support for WKInterfaceDevice, and class methods in WKInterfaceController

### PR DESCRIPTION
**Pull request by gudatcomputers**
So our project was using the `openParentApplication:reply:` selector and we had to fork it to allow our specs to compile.  I sat down and tried to figure our how to test these class methods without adding support for class stubs in Cedar.  I spoke with Sam Coward about this briefly and without doing some more gnarly implementation of a custom Cedar matcher or NSInvocation category, this is what I came up with.  The tests pass for `openParentApplication:reply:` and `reloadRootControllersWithNames:contexts:`

My primary motivation for submitting this is not wanting to be on a custom fork of PCK for our Watch project.  Obviously making it available for others is important as well, but I didn't want us to miss out on future PCK updates.

### CODE REVIEW

**Comment by gudatcomputers** [comment]
I managed to get this working as an `NSInvocation` category as well.  If the maintainers prefer this method I can do that instead.  Also used a `__bridge_retained` cast here (as I probably should have in the original commit).  I noticed that calling `matchesSelector:withArguments:`  for the same invocation and argument array in sequence was producing EXC_BAD_ACCESS and through some zombies and trial and error I realized it was coming from the array of arguments being passed to the category method. Switching to `__bridge_retained` fixes the EXC_BAD_ACCESS but I don't know that this approach is actually "proper"
screen shot 2015-04-12 at 2 33 06 pm

**Comment by pivotal-brian-croom** [comment]
You would probably be best off declaring retrievedArg as `__unsafe_unretained id` and removing the bridging cast. This will effectively disable ARC for that local variable, which is helpful because NSInvocation doesn't touch the retain count when copying bytes in the getArgument:atIndex: call.

**Comment by gudatcomputers** [comment]
Thanks Brian,

Appreciate the tip on the declaration.  Would you prefer this be implemented  as the `NSInvocation` category shown above?

**Comment by gudatcomputers** [comment]
@pivotal-brian-croom Any thoughts? Would you prefer the `NSInvocation` category?  I have it working with the updated `__unsafe_unretained id` like you mentioned/

**Comment by akitchen** [comment]
hey thanks so much for the pr, @gudatcomputers  / @mckeejm 

Good advice from @pivotal-brian-croom as always.  ARC needs to be told not to try to manage the objc object retrieved from the NSInvocation in this case.

Can you fix up and re-push?  I'm fine with the NSInvocation category if you think it helps.  However, in order to be reusable it should maybe also check the invocation's target.

**Comment by gudatcomputers** [comment]
Thanks @akitchen I'll add the check for invocation target, and re-push the commit.

**Comment by gudatcomputers** [comment]
Went ahead and appended the next item I was working on to this PR since it was so small.  The second commit adds support for WKInterfaceDevice

**Comment by gudatcomputers** [comment]
@akitchen anything else you want changed about this? I'm anxious to get this PR resolved and get off the fork :)

File: WatchKit/WatchKit/NSInvocation+Spec.m [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    if (selectorArgumentCount != arguments.count) {
+        return NO;
+    }
+
+    for(NSInteger idx = 0; idx < selectorArgumentCount; idx++) {
+        id argument = [arguments objectAtIndex:idx];
+        __unsafe_unretained id retrievedArg;
+        [self getArgument:&retrievedArg atIndex:idx + 2];
+        if(retrievedArg != argument) {
```
Comment (akitchen) [inline WatchKit/WatchKit/NSInvocation+Spec.m]: What about also checking object equality here?

**Comment by akitchen** [comment]
For the invocation matcher category to be reusable on its own, I'd like to see it support object equality for the arguments as well as pointer comparison.  Also, could you rename the category from Spec to "InvocationMatching" or some such?  I'd prefer categories such as these to have names that reflect what support they add.

Otherwise, I think we're good to go, and subsequent changes/additions can then happen on master.

Thanks again @gudatcomputers

**Comment by gudatcomputers** [comment]
So you mean the argument matching would be updated to `pointersAreEqual || objectsAreEqual` ?  Sounds good to me, I'll rename the category and push those changes.  I feel like the category should move out of WatchKit if we're targeting it for larger reuse?

**Comment by akitchen** [comment]
Yes, it will likely move. But we can make that change later, on master, when needed.




Thanks again

**Pull request merged**